src/pools.jl

```diff
@@ -51,6 +51,29 @@ safesizehint!(x, n) = sizehint!(x, min(4096, n))
 # determines whether we'll look up object caches in .keyedvalues or .values
 iskeyed(::Pool{K}) where {K} = K !== Nothing
 
+"""
+    Pools.max(pool::Pool) -> Int
+
+Return the maximum number of objects permitted to be in use at the same time.
+See `Pools.permits(pool)` for the number of objects currently in use.
+"""
+max(pool::Pool) = Base.@lock pool.lock pool.max
+
+"""
+    Pools.permits(pool::Pool) -> Int
+
+Return the number of objects currently in use. Less than or equal to `Pools.max(pool)`.
+"""
+permits(pool::Pool) = Base.@lock pool.lock pool.cur
```

Comment:
maybe rename this `usage` or something like that

Reply:
`in_use()`?

Reply:
I agree i don't like the name `permits`. I would have assumed this to be `max capacity - in use` (i.e. the number of permits still available)